lib/screens/home/home_view_model.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:lydiaryanfluttersurvey/model/response/surveys_response.dart';
+import 'package:lydiaryanfluttersurvey/model/ui/survey_ui_model.dart';
+import 'package:lydiaryanfluttersurvey/screens/home/home_state.dart';
+import 'package:lydiaryanfluttersurvey/usecases/base/base_use_case.dart';
+import 'package:lydiaryanfluttersurvey/usecases/get_surveys_use_case.dart';
+import 'package:rxdart/subjects.dart';
+
+class HomeViewModel extends StateNotifier<HomeState> {
+  final GetSurveysUseCase _getSurveysUseCase;
+
+  final BehaviorSubject<List<SurveyUiModel>> _surveys = BehaviorSubject();
+  Stream<List<SurveyUiModel>> get surveys => _surveys.stream;
+
+  final BehaviorSubject<String> _error = BehaviorSubject();
+  Stream<String> get error => _error.stream;
+
+  HomeViewModel(this._getSurveysUseCase) : super(const HomeState.init());
+
+  void getSurveys() async {
+    final result = await _getSurveysUseCase.call();
```

Comment:
Do we miss updating the state to `HomeState.loading()` here? 🤔

Reply:
In the first implementation, setting the state to `loading` immediately caused the app to crash 🙈 Currently I have removed the loading state and just use the `init` state to show the loading shimmer 🙏🏻